### PR TITLE
DO NOT MERGE: Smallfile change log parsing

### PR DESCRIPTION
**Pull request by Avilir**
Changing the way results are pars. instead of reading results from elasticsearch (since it is not pushing results to ES when the test run in the public cloud), the results are parse from the log(s) file(s).
It also support running with more then one pod (for scaling).

One more change is the samples deviation results from 5% to 20% - this will changed when we will have hardware and network infra. that will run the test properly.

The test results are print in the log and also push to Internal elasticsearch server, in the future it will be push to codespeed server.

### CODE REVIEW

File: tests/e2e/performance/test_small_file_workload.py [hunk 1]
```diff
@@ -17,11 +17,138 @@
...
 from ocs_ci.ocs.exceptions import CommandFailed
-from elasticsearch import (Elasticsearch, exceptions as ESExp)
+from elasticsearch import (Elasticsearch, exceptions as esexp)
 import numpy as np
 
 log = logging.getLogger(__name__)
 
+"""
+Defining dictionary of keywords to look in the logs, foreach keyword we have the
+key-name in the final results output, the seperate charecter to split the line
```
Comment (fbalak) [inline tests/e2e/performance/test_small_file_workload.py:27]:
```suggestion
key-name in the final results output, the separate character to split the line
```

File: tests/e2e/performance/test_small_file_workload.py [hunk 2]
```diff
@@ -17,11 +17,138 @@
...
+    'total data': {
+        'name': 'Total_Data_GB', 'sep': ' ', 'pos': -2, 'op': np.sum
+    },
+}
+
+
+class SmallFileLogParser(object):
+
+    """
+    This class is reading the benchmark pod logs, and pars it
```
Comment (fbalak) [inline tests/e2e/performance/test_small_file_workload.py:67]:
```suggestion
    This class is reading the benchmark pod logs, and parse it
```

File: tests/e2e/performance/test_small_file_workload.py [hunk 3]
```diff
@@ -105,152 +235,51 @@ def add_key(self, key, value):
...
-
-        results = {}
-
-        for op in self.results['operations']:
-            for smp in range(self.results['samples']):
-                sample = smp + 1
-                if op in self.results['full-res'].keys():
-                    self.results['full-res'][op][sample] = self.combine_results(
-                        self.results['full-res'][op][sample], True)
+            for op in self.results['operations']:
```
Comment (fbalak) [inline tests/e2e/performance/test_small_file_workload.py:271]: Can you please add docstring explaining the process here?

File: tests/e2e/performance/test_small_file_workload.py (hunk 1 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py]: done.

File: tests/e2e/performance/test_small_file_workload.py (hunk 2 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py]: done

File: tests/e2e/performance/test_small_file_workload.py (hunk 3 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py:278]: done

File: tests/e2e/performance/test_small_file_workload.py [hunk 4]
```diff
@@ -424,64 +439,93 @@ def test_smallfile_workload(self, ripsaw, file_size, files, threads,
...
-                if bench_pod[0] is not None:
-                    small_file_client_pod = bench_pod[0]
-                    break
-            except IndexError:
-                log.info("Bench pod not ready yet")
+        clients = sf_data['spec']['workload']['args']['clients']
+        log.info(f'Going to run on {clients} pods, waiting for pods to start')
+        count = 10  # total number of retry for all benchmark pods to start
+        small_file_client_pods = {}
+        while count > 0:
```
Comment (vasukulkarni) [inline tests/e2e/performance/test_small_file_workload.py:462]: do we need the count, how about increasing the values in timeout sampler? will that help?

File: tests/e2e/performance/test_small_file_workload.py [hunk 5]
```diff
@@ -105,152 +235,77 @@ def add_key(self, key, value):
...
+            operations that we have in this particular test.
 
-        Returns:
-            dict : dictionary of results records
+            """
+            for op in self.results['operations']:
 
-        """
+                """
+                For each operation, I am looking for all interesting lines
```
Comment (vasukulkarni) [inline tests/e2e/performance/test_small_file_workload.py:281]: I think better to add small snipet of items we are parsing here.

File: tests/e2e/performance/test_small_file_workload.py (hunk 4 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py]: changed the timeout, instead using the count.

File: tests/e2e/performance/test_small_file_workload.py (hunk 5 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py:317]: added example of the log file.

**Comment by Avilir** [comment]
I am working on different approach, so, pleas wait with this PR

**Comment by Avilir** [comment]
replaced by PR #2397

**Pull request closed without merging**